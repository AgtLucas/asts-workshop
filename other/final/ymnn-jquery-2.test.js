import stripIndent from 'strip-indent'
import * as recast from 'recast'
import * as babel from 'babel-core'
import ymnnJquery from './ymnn-jquery-2'

test('transpiles jquery calls to raw DOM APIs', () => {
  const source = stripIndent(
    `
      $(el).hide()
      foo.hide()

      $(el).show()
      bar.show()
    `,
  ).trim()
  const code = transpile(source)
  expect(code).toMatchSnapshot()
})

function transpile(source) {
  const {code} = babel.transform(source, {
    parserOpts: {parser: recast.parse},
    generatorOpts: {generator: recast.print},
    babelrc: false,
    plugins: [ymnnJquery],
  })
  return code
}
