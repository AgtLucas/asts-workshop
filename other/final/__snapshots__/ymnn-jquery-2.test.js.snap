// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transpiles jquery calls to raw DOM APIs 1`] = `
"el.style.display = \\"none\\"
foo.hide()

el.style.display = \\"\\"
bar.show()"
`;
